firebase-vscode/src/core/config.ts

```diff
@@ -104,112 +112,207 @@ function notifyFirebaseConfig(broker: ExtensionBrokerImpl) {
   });
 }
 
-function registerRc(broker: ExtensionBrokerImpl): Disposable {
-  firebaseRC.value = _readRC();
-  const rcRemoveListener = onChange(firebaseRC, () =>
-    notifyFirebaseConfig(broker),
-  );
+function displayStringForUri(uri: vscode.Uri) {
+  return vscode.workspace.asRelativePath(uri);
+}
 
-  const showToastOnError = effect(() => {
-    const rc = firebaseRC.value;
-    if (rc instanceof ResultError) {
-      vscode.window.showErrorMessage(`Error reading .firebaserc:\n${rc.error}`);
-    }
+function notifyFirebaseConfigListChanged(broker: ExtensionBrokerImpl) {
+  broker.send("notifyFirebaseConfigListChanged", {
+    values: allFirebaseConfigsUris.value.map(displayStringForUri),
+    selected: selectedFirebaseConfigUri.value
+      ? displayStringForUri(selectedFirebaseConfigUri.value)
+      : undefined,
   });
+}
 
-  const rcWatcher = _createWatcher(".firebaserc");
-  rcWatcher?.onDidChange(() => (firebaseRC.value = _readRC()));
-  rcWatcher?.onDidCreate(() => (firebaseRC.value = _readRC()));
-  // TODO handle deletion of .firebaserc/.firebase.json/firemat.yaml
-  rcWatcher?.onDidDelete(() => (firebaseRC.value = undefined));
+async function registerRc(
+  context: vscode.ExtensionContext,
+  broker: ExtensionBrokerImpl,
+) {
+  let disposable: Disposable | undefined;
+  context.subscriptions.push({ dispose: () => disposable?.dispose() });
```

Comment:
This can be combined into the effect return value

On line 156:

`return () => disposable?.dispose();`